### ISSUE TITLE
Optionally disable the sleep/wake up cycle in namespace

### ISSUE THREAD

**Issue by davidebianchi** [kube-green/kube-green#76]
The SleepInfo CRD could have a new optional field, which enable or disable the controller functionality.

It could be useful not to totally remove the CRD, but disable it. For example, if for a specific period of time the selected namespace should not be sleeped.

**Comment by sswastik02** [comment]
Hi @davidebianchi ,
Is this issue still relevant?
If yes, I would like to work on it.

**Comment by davidebianchi** [comment]
It could be still relevant, but it could also be related with https://github.com/kube-green/kube-green/issues/332 since it allow to suspend the lifecycle for some day so maybe we already achieved the goal.

**Comment by sswastik02** [comment]
Hi @davidebianchi,

You're correct. Once implemented, it enables us to temporarily suspend the lifecycle for certain days by forcing sleep or maintaining activity on specific days.

> The SleepInfo CRD could have a new optional field, which enable or disable the controller functionality.

However, I believe the main concept here is to facilitate the enabling or disabling of the CRD itself. This functionality would allow us to toggle the controller's operation at any arbitrary point in time. 

What do you think about it?

BR

**Comment by davidebianchi** [comment]
Do you intend to add a `suspend` field inside the CRD? Sorry for the late reply

**Comment by sswastik02** [comment]
Yes @davidebianchi, it would suspend the functioning of the CRD, much like flipping a switch.

**Comment by nemcikjan** [comment]
@davidebianchi Hey, any progress on this? We're looking to utilize this project but we'd favor a "one-click" functionality to temporarily enable/disable the cycle to allow manual intervention if cluster would need to be used occasionally also during "sleep" hours. If any help is needed, I may jump in and help implement the feature. I can also imagine an enhanced version driven by some Prometheus metrics. Looking forward to hear back.

**Comment by davidebianchi** [comment]
Hi @nemcikjan! Do you need a way to wake up a stopped namespace? Or  only to suspend the SleepInfo lifecycle?

**Comment by nemcikjan** [comment]
> Hi [@nemcikjan](https://github.com/nemcikjan)! Do you need a way to wake up a stopped namespace? Or only to suspend the SleepInfo lifecycle?

Preferably both, but suspending the SleepInfo is more important at the moment for us. I'm able to find a way to wake up stopped namespace on my own, but if SleepInfo will be able to do it, that'd be preferred.
